content/hypi-directives.md

```diff
@@ -0,0 +1,550 @@
+      "extensions": {}
+    }
+  ]
+}
+```
+</TabItem>
+</Tabs>
+
+### @computed
+
+This directive allows the value of a field to be calculated using an ArcQL query. You may also use a groovy-like function to combine the values in the fields. You may combine string or date values to match your requirements. 
+
+`combine: String@computed(postQueryFn: """self.hypi.id + ":" + self.title """)`
+
+<Tabs
+  defaultValue="query"
+  values={[
+    {label: 'GraphQL Query', value: 'query'},
+    {label: 'Response', value: 'response'},
+  ]}>
+<TabItem value="query">
+
+```java
+{
+  find(type: Post, arcql: "*") {
+    edges {
+      node {
+        ... on Post {
+           title
+           postCreated
+           augmentedTitle
+           date
+           combine
+        }
+      }
+      cursor
+    }
+  }
+}
+```
+
+</TabItem>
+<TabItem value="response">
+
+```json
+{
+  "data": {
+    "find": {
+      "edges": [
+        {
+          "node": {
+            "title": " Hi There!",  // Upserted value of title
+            "postCreated": "2021-04-12T05:23:37Z",
+            "augmentedTitle": " Hi There! computed",
+            "date": "2021-04-12T05:25:10.922219Z",
+            "combine": "01F328J1B9BFMVBY8G63R6KXWS: Hi There!"
+          },
+          "cursor": "01F328J1B9BFMVBY8G63R6KXWS"
+        }
+      ]
+    }
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+### @noinput / @nomath / @noaggs / @nomath
+
+Hypi automatically generates certain data types when you declare a type. These generated types are used to implement functionalities like CRUD, Aggregations, Maths functions, etc.
+
+Let’s say you declared the following data type.
+```java
+typeAWithNoDirs {
+    f1: String
+}
+```
+When you save the schema, the following tables will be automatically generated.
+```java
+AWithNoDirs
+AWithNoDirsInputOpt
+AWithNoDirsAggs
+AWithNoDirsGroupByOptions
+AWithNoDirsInput
+HypiUpsertInputUnion.AWithNoDirs: [AWithNoDirsInputOpt!]
+```
+You may check the details of these types under the documentation on the Hypi platform.
+
+Now, declare the same data type like this:
+```java
+typeAWithNoDirs @noagg @notable @noinput @nomath {
+    f1: String
+}
+```
+When you save the above data type, only the AWithNoDirs table will get generated. Other tables will not be generated and the user will not be able to perform associated functions.
+
+@noinput tells Hypi not to generate an input type for this type. This prevents objects of this type from being used in upsert and other functions. So, you may not be able to perform mutations on this data type.
+
+@notable tells Hypi not to generate a table for the type it is attached to (AWithNoDirs here). Instead, the user will use a serverless function or some other method to produce objects of this type. So, you may not be able to create objects of this type.
+
+@noagg Tells Hypi not to generate aggregation fields for the type this is attached to. So you may not be able to perform aggregation functions on AWithNoDirs
```

Comment:
Here and below you say `may not` but it needs to be `will not` because there isn't some way to do it once these directives are used

Reply:
corrected